From d047f90b0de01c2a9cd943c86c912b6555ad7393 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:37 -0400
Subject: [ethernet] mlx4: fix error return code in mlx4_en_get_qp()

Message-id: <e1c6faa8da2f8b62c0b3e7688a8b2f03ce67be14.1410425016.git.avadai@redhat.com>
Patchwork-id: 93620
O-Subject: [Patch RHEL7.1 V1 041/197] net/mlx4_en: fix error return code in mlx4_en_get_qp()
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>

Fix to return a negative error code from the error handling
case instead of 0.

Fixes: 837052d0ccc5 ('net/mlx4_en: Add netdev support for TCP/IP offloads of vxlan tunneling')
Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 9ba75fb0c4b92416b94640b5a043c323a457f14a)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
index 60b927e..fad4531 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
@@ -632,8 +632,9 @@ static int mlx4_en_get_qp(struct mlx4_en_priv *priv)
  if (err)
   goto steer_err;
 
- if (mlx4_en_tunnel_steer_add(priv, priv->dev->dev_addr, *qpn,
-         &priv->tunnel_reg_id))
+ err = mlx4_en_tunnel_steer_add(priv, priv->dev->dev_addr, *qpn,
+           &priv->tunnel_reg_id);
+ if (err)
   goto tunnel_err;
 
  entry = kmalloc(sizeof(*entry), GFP_KERNEL);
-- 
1.7.1